datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -557,18 +557,16 @@ impl BinaryExpr {
         match &self.op {
             IsDistinctFrom | IsNotDistinctFrom | Lt | LtEq | Gt | GtEq | Eq | NotEq
             | Plus | Minus | Multiply | Divide | Modulo => unreachable!(),
-            And => {
-                if left_data_type == &DataType::Boolean {
-                    boolean_op!(&left, &right, and_kleene)
-                } else {
-                    internal_err!(
-                        "Cannot evaluate binary expression {:?} with types {:?} and {:?}",
-                        self.op,
-                        left.data_type(),
-                        right.data_type()
-                    )
-                }
-            }
+            And => match left_data_type {
+                DataType::Boolean => boolean_op!(&left, &right, and_kleene),
+                DataType::List(_field) => array_intersect(&[left, right]),
```

Comment:
I am not sure if DataFusion already has some stuff to handle function -> operator aliasing. At least, we should parse/convert the `AND` operator to `array_intersect` function call at early stage, instead of silently turning to call `array_intersect` in binary physical operator. If we are going to support more aliasing operators like this, we will soon find these aliasing logic scatters everywhere in codebase and hard to track.

Reply:
I think this is a good suggestion. Like what we have done for MakeArry in `sql_array_literal`. I think we can do the similar conversion for other List operator in sql to logicExpr stage. Including array_concat, append prepend, array_has

Reply:
how about we make this change in the following PR? it seems that we need to refactor `array_concat, append prepend, array_has` too